cpp/src/copying/contiguous_split.cu

```diff
@@ -742,6 +746,32 @@ struct dst_offset_output_iterator {
   reference __device__ dereference(dst_buf_info* c) { return c->dst_offset; }
 };
 
+/**
+ * @brief Output iterator for writing values to the valid_count field of the
+ * dst_buf_info struct
+ */
+struct dst_valid_count_output_iterator {
+  dst_buf_info* c;
+  using value_type        = size_type;
+  using difference_type   = size_type;
+  using pointer           = size_type*;
+  using reference         = size_type&;
+  using iterator_category = thrust::output_device_iterator_tag;
+
+  dst_valid_count_output_iterator operator+ __host__ __device__(int i)
```

Comment:
This operator only ever gets used during a single thrust device algorithm, right? So it doesn't need to be `__host__`?

Reply:
No, it ends up getting called on the host, although it's not entirely clear when.  If you leave them marked just as __device__, you get a hard crash.